sensor/common/sensor/central_communication.go

```diff
@@ -22,7 +22,6 @@ func NewCentralCommunication(components ...common.SensorComponent) CentralCommun
 		sender:     NewCentralSender(components...),
 		components: components,
 
-		stopC:    concurrency.NewErrorSignal(),
-		stoppedC: concurrency.NewErrorSignal(),
+		stopper: concurrency.NewStopper(),
```

Comment:
Looking at the changes on this component and the others, my understanding is that each component has its own Stopper instance and external shutdown requests should come from `Stop` function. 

This is a bit conflicting with my understanding of the responsibility of `Flow` and `Client` (as per I commented above). I thought the `Client` was to be used by a parent caller (e.g. sensor's main goroutine), and the `Flow` by each component?

If `Stopper` contained in a single component's lifecycle, then I don't understand why we need these two interfaces. Why not have all the methods in the same one then? It seems to make the client code more difficult to reason about. Example:

```go
s.stopper.Flow().StopWithError(err)
// vs
s.stopper.StopWithError(err)

s.stopper.Client().Stop()
// vs
s.stopper.Stop()
```

Reply:
It could have been one interface, but the problem is that I saw a number of misinterpretations and problems when the pattern was implemented with two (Error)Signal-s. Forcing separation of interfaces is intentional to help spot when wrong things are used in wrong places.

`StopWithError` is part of `Flow` because I found that reporting stop with error only makes sense from within the goroutine, but not outside of it.